Add additionalEventData to login response & make onHookShouldAwait configurable.

B77Mills opened the pull request:
Add additionalEventData prop to login response, not just spread it onto the response.

B77Mills committed:
Put additionalEventData to login response and not just spread it

Add additionalEventData prop to login response, not just spread it onto the response.

B77Mills committed:
add onHookShouldAwait props to be configured

Allow for should await to be configure from within site’s configuration fo omedaIdX

In the case of AB await loginLinkSent in order to auto signup users.

solocommand commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Gate all 'Article' content types for AQUA only](https://trello.com/c/17UyjmlC/446-gate-all-article-content-types-for-aqua-only)

zarathustra323 commented on packages/marko-web-omeda-identity-x/add-integration-hooks.js:
```diff
@@ -44,6 +47,9 @@ module.exports = (params = {}) => {
     omedaPromoCodeCookieName: Joi.string().required(),
     omedaPromoCodeDefault: Joi.string(),
+    onLoginLinkSentShouldAwait: Joi.boolean().default(false),
+    onAuthenticationSucessShouldAwait: Joi.boolean().default(true),
```
Typo here in `success`

zarathustra323 commented on packages/marko-web-omeda-identity-x/add-integration-hooks.js:
```diff
@@ -29,6 +29,9 @@ module.exports = (params = {}) => {
     idxOmedaRapidIdentifyProp,
     omedaPromoCodeCookieName,
     omedaPromoCodeDefault,
+    onLoginLinkSentShouldAwait,
```
Instead of making these individual props, can we move this into a single object called `shouldAwait`?

```js
shouldAwait: Joi.object({
  onLoginLinkSent: Joi.boolean().default(false),
  onAuthenticationSuccess: Joi.boolean().default(false),
  // ... and others
}).default(),
```

The just pass the corresponding value:
`shouldAwait: shouldAwait.onLoginLinkSent`

zarathustra323 commented on packages/marko-web-omeda-identity-x/index.js:
```diff
@@ -98,6 +101,9 @@ module.exports = (app, params = {}) => {
     omedaPromoCodeCookieName: Joi.string().default('omeda_promo_code'),
     omedaPromoCodeDefault: Joi.string(),
     omedaRapidIdentifyProp: Joi.string().default('$omedaRapidIdentify'),
+    onLoginLinkSentShouldAwait: Joi.boolean().default(false),
```
Same here... perhaps make this a re-usable Joi schema and import it into both files

B77Mills committed:
make a shared should-await schema

Convert the individual onLoginSentShouldAwait, onAuthenticationSussessShouldAwait & onUserProfileUpdateShouldAwait into a shared should-await schema.
